### PR TITLE
Add support for creating partial indexes on tables

### LINKED ISSUE #322: Support for partial indexes
Our team tried to use pgroll for our new project and found out that it doesn't seem to have a partial index support. 

It seems there were mentions while adding support for indexes (#49) but seems like it is not still implemented.

Since our new project highly depends on partial index, we're wondering if there is any plan for adding this feature.

### PR DESCRIPTION
**Pull request by kvch**
This PR adds a new option to `create_index` operations called `predicate`. When `predicate` is the conditional expression of the partial index. When it is set, pgroll creates a partial index on the table.

Example:
```json
{
  "name": "37_create_partial_index",
  "operations": [
    {
      "create_index": {
        "name": "idx_fruits_id_gt_10",
        "table": "fruits",
        "columns": [
          "id"
        ],
        "predicate": "id > 10"
      }
    }
  ]
}
```

Closes #322

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -963,13 +963,15 @@ A create index operation creates a new btree index on a set of columns.
...
     "columns": [ "names of columns on which to define the index" ]
+    "predicate": "conditional expression for defining a partial index",
   }
 }
 ```
 
 Example **create index** migrations:
 
 * [10_create_index.json](../examples/10_create_index.json)
+* [37_create_partial_index.json](../examples/37_create_partal_index.json)
```
Comment (ryanslade) [inline docs/README.md:974]:
```suggestion
* [37_create_partial_index.json](../examples/37_create_partial_index.json)
```

File: schema.json [hunk 2]
```diff
@@ -223,6 +223,10 @@
         "table": {
           "description": "Name of table on which to define the index",
           "type": "string"
+        },
+        "predicate": {
```
Comment (exekias) [inline schema.json:227]: I'm wondering if this would be better called `where`, to avoid increasing cognitive load on the user

Comment (kvch) [reply schema.json:227]: I am open to changing it to anything that makes it clearer to the user.

**Review by andrew-farries — COMMENTED** [review]
This looks good 👍 

We should also update the `read_schema` SQL function to return information about the predicate of a parital index. Right now the new index from the example looks like this:

```json
{
  "indexes": {
    "idx_fruits_id_gt_10": {
      "name": "idx_fruits_id_gt_10",
      "unique": false,
      "columns": [
        "id"
      ]
    }
  }
}

```

(this is part of the `resulting_schema` field after the migration to add the partial index has been applied).

File: schema.json (hunk 2 above)
Comment (exekias) [reply schema.json:227]: ah just saw this is actual postgres nomenclature in the docs, so we are good

Comment (andrew-farries) [reply schema.json:227]: I think 'predicate' is OK as that is the term used in the Postgres [docs on partial indexes](https://www.postgresql.org/docs/current/indexes-partial.html).

I'm fine with either `where` or `predicate` personally.

Comment (exekias) [reply schema.json:227]: 👍 let's use `predicate`

**Comment by kvch** [comment]
Updated the PR so the output of `analyze` contains the predicate as well:
```json
"idx_fruits_id_gt_10": {
  "name": "idx_fruits_id_gt_10",
  "unique": false,
  "columns": [
    "id"
  ],
  "predicate": "(id \u003e 10)"
}
```

**Review by andrew-farries — APPROVED** [review]

**Pull request merged**